File: python/tree.py
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def insert(root, value):
    if root is None:
        return TreeNode(value)
    if value < root.value:
        root.left = insert(root.left, value)
    else:
        root.right = insert(root.right, value)
    return root


def in_order_traversal(root):
    if root:
        in_order_traversal(root.left)
        print(root.value, end=" ")
        in_order_traversal(root.right)


def find(root, value):
    if root is None:
        return False
    if value == root.value:
        return True
    elif value < root.value:
        return find(root.left, value)
    else:
        return find(root.right, value)


def run_tree_demo():
    print("\n🌳 Tree Demo")
    values = [5, 3, 7, 2, 4, 6, 8]
    root = None
    for v in values:
        root = insert(root, v)

    print("In-order Traversal:", end=" ")
    in_order_traversal(root)
    print()

    search = 6
    print("Contains", search, "?", find(root, search))

    search = 10
    print("Contains", search, "?", find(root, search))

File: csharp/Tree.cs
using System;

public class TreeNode
{
    public int Value;
    public TreeNode? Left;
    public TreeNode? Right;

    public TreeNode(int value)
    {
        Value = value;
        Left = null;
        Right = null;
    }
}


public static class TreeDemo
{
    public static TreeNode Insert(TreeNode? root, int value)
    {
        if (root == null)
            return new TreeNode(value);
        if (value < root.Value)
            root.Left = Insert(root.Left, value);
        else
            root.Right = Insert(root.Right, value);
        return root;
    }

    public static void InOrderTraversal(TreeNode? node)
    {
        if (node == null) return;
        InOrderTraversal(node.Left);
        Console.Write(node.Value + " ");
        InOrderTraversal(node.Right);
    }

    public static bool Find(TreeNode? root, int value)
    {
        if (root == null) return false;
        if (value == root.Value) return true;
        else if (value < root.Value) return Find(root.Left, value);
        else return Find(root.Right, value);
    }


    public static void RunTreeDemo()
    {
        Console.WriteLine("\n🌳 Tree Demo");
        int[] values = { 5, 3, 7, 2, 4, 6, 8 };
        TreeNode? root = null;
        foreach (var v in values)
            root = Insert(root, v);

        Console.Write("In-order Traversal: ");
        InOrderTraversal(root);
        Console.WriteLine();

        int search = 6;
        Console.WriteLine($"Contains {search}? {Find(root, search)}");

        search = 10;
        Console.WriteLine($"Contains {search}? {Find(root, search)}");

    }
}
